### ISSUE TITLE
Icarus

### ISSUE THREAD

**Issue by TwinNumberOne** [siggame/MegaMinerAI-Game-Ideas#64]
**TLDR**
-
- this is for mega miner
- You mine resources from floating islands
- your ships are fixed but can use multiple ammo types
- you can move cargo between ships
- Your goal is to destroy the enemy base or to have lost the least at the end of the game.

**Story**
-
Two empires have finally sparked into open war and two generals are fighting a key battle to see who can control a resource-rich area of land and a lift crystal mine. A rare resource that when put under pressure generates extra lift, but disintigrates when it takes too much damage.

**The Game**
-
**Victory**
Destroy the enemy base or have mined the most lift crystal by the end of the game

**Map**
- a 100x100 map, littered with floating islands, pockets of float crystal dust (which will vanish if hit by flame) 
- With each base being one plateau's on each side. 
- The floating islands are able to be moved by the airships, and the dust is able to be vacuumed up
- the dust reduces all kinetic damage going through it (if you want you can make it do less the more it goes through or just a flat decrease)
- All floating islands have ore in them and a lift crystal core, where depending on the size it floats higher or lower.
- if islands are hit and damaged they release lift crystal dust.

**Movement**
movement costs power which is fueled by their steam engines that consume coal (which can burn), or lift crystal (which explodes, produces more power than coal). Airships will have to avoid floating islands, and flying through float crystal dusk clouds around the map with floating islands providing cover from your enemy. Also, ships have there float ceiling reduced based on the amount of cargo they have

**Resource Collection**
All floating islands have a resource value and take up multiple tiles, they can be mined in two ways they can be moved into the mining area of your base by airship with a hook and or by being pushed to be mined. Or you can have an airship go and mine the floating island out in the field, but should it mine lift crystal from it the island will start sinking to the ground. Otherwise, lift crystal dusk clouds can be mined by putting a vacuum on your ship to collect lift crystal dust. The miners can collect, ore, lift crystal, and coal.

**Unites**
cost is (ore/lift crystal) All unites need fuel to fuel their engines and they can use coal, oil, and lift the crystal, in order of lest volatile to most, from gives less power to gives the most power.
everything will be ranked in cost by 1-10. so, for instance, a 10 in float ceiling would be 120 blocks of hight.

| Name | Cost | HP | Cargo hold | float ceiling | speed |
|----|----|----|----|----|----|
|**Miner**                 |5/5|1|8|10|8|
|**Gunboat**             |8/8|5|5|9|6| 
|**Ironclad**              |10/10|10|7|8|4| 
|**Frigate**                  |7/7|7|5|9|7| 
|**Destroyer**             |9/9|3|6|8|5| 
|**Tanker**                  |10/10|3|10|10|4| 
|**Corvette**               |1/1|1|3|10|10|

**Unite info**
All units can move cargo between each other they need to be right next to each other to do it though.
__Miner__
This unite when parked next to a floating island can mine resources from it, this will decrease the weight of the island making it float higher and the ore will weight down the miner, meaning mining needs to be managed to stop the island from floating too high, but this can also be used to crush an enemy ship or to move the islands to a more desired location. Though shooting islands with ballistic rounds or hitting them with missiles will also reduce their weight and make them float higher, but get rid a bit ore as islands aren't pure ore and float crystal.
__Gunboat__
This unite needs to manage its fuel stores and ammo stores to make sure it stays a threat as due to its low cargo hold, it deals decent damage but has a hard time getting through a shield (does reduce damage to the shield.
__Ironclad__
Using float crystal technology they weaponized how the float crystal clouds work to produce a shield that can be fired out of but reduces in size as its hp decreases, it covers an 11x11 area at max health.
__Destroyer__
Best known for its use of missiles and torpedos to blow through shields and ships alike, but it has paper-thin armor and will fall to even a gunboat and ignite under a frigate.
__Tanker__
it's slow and has a large cargo hold use this to bring what you need to keep your fleet up.
__Corvette__
Fast, large cargo hold for its size used to deliver resources to ships and otherwise move them around. It has very weak armor and will light up like a firework if hit by an explosion.

**Combat**
Torpedos beat shields, guns does reduced damage to shields, and normal to heavily armored targets. Flame does nothing to shields but burns up ammo stores.
Floating islands can stop you from being shot. All states of the ammo listed below are out of 10 and a 10 in the range should be able to reach 1/3 of the map in optimal circumstances. missiles take 3 rounds to reach their max range. Mines aren't too expensive and can be easily shot down.

| Name | Cost | damage | Range | Cargo cost | special |
|----|----|----|----|----|----|
|**basic shot**                 |2|3|6|1|normal|
|**far shot**                 |4|3|8|1|goes further than basic|
|**percing shot**                 |4|6|6|1|does more damagel|
|**explosive shot**                 |7|8|5|3|can ignite stuff like a flame thrower|
|**scatter shot**                 |5|2|3|2|hits enemies in a 4x4 area|
|**Torpedo**                 |8|8|10|3|goes steight, can be shot down|
|**crystal cloud**                 |2 mineral 5 crystal|0|10|3|a torpedo that makes a crystal cloud|
|**Mine**                 |10|10|1|3|it floats in a spot and then rockets to an enemy ship that passes by|

**Remarks**
So you got your basic rock paper scissors combat with a shield that blocks stuff, a missile boat that can get through it where its rounds can be shot down. But then you can have fun with moving around the floating islands (that can move!) and dealing with different ammo types (of which the exploding and burning of lift crystal is easily cut as that could be too much). Along with making sure your units are supplied with plenty of fuel and ammo.
also maybe add bombs, but with all the ammo types I feared it might be too much...
I have gotten rid of a feature of an alternate fuel of oil and flame throwers and fire, it just seems to make it too complicated.
Otherwise If you need to cut anything else the ammo types would be easy enough to cut I just like the idea of there being able to be minefields, and unites utilizing different ammo types, makes it more alive.
I do like the idea of the exploding crystal clouds as it could be weaponized as another way to get around sheilds but these programmers only have so much time.

**Comment by TwinNumberOne** [comment]
please put this for MegaMinerAI-26
